packages/graph/hash_graph/lib/graph/src/store/postgres/context/entity.rs

```diff
@@ -21,7 +21,9 @@ pub struct EntityRecord {
 
 pub type RecordStream = impl Stream<Item = Result<EntityRecord, QueryError>>;
 
-fn row_stream_to_record_stream(row_stream: RowStream) -> RecordStream {
+fn row_stream_to_record_stream(
+    row_stream: RowStream,
+) -> impl Stream<Item = Result<EntityRecord, QueryError>> {
```

Comment:
Did clippy suggest this change?

Reply:
With the newest version this didn't compile anymore (which is correct, it's an opaque type which is specified below).

Reply:
Is there a point in us still having 
```
pub type RecordStream = impl Stream<Item = Result<EntityRecord, QueryError>>;
```
Then?

Reply:
Yes, we need it as the return type for a trait. Otherwise, we would need to write our own stream.

It's used e.g. in
```rust
#[async_trait]
impl<C: AsClient> PostgresContext for PostgresStore<C> {
    async fn read_all_ontology_types<T>(&self) -> Result<ontology::RecordStream<T>, QueryError>
    where
        T: OntologyDatabaseType + TryFrom<serde_json::Value, Error: Context>,
    {
        ontology::read_all_types(&self.client, T::table())
            .await
            .attach_printable("could not read ontology types")
    }
```

Reply:
Oh is this related to [this](https://github.com/rust-lang/rust/pull/101224)?

Reply:
No, but we *could* use RPITIT from now on (if it wouldn't result in ICEs all the time 😆)

Reply:
But, yes, in theory we could use RPITIT here as soon as it's more stable 🙂